### ISSUE TITLE
[Bug][Jupyter Widget] Hexbin fails with EnhancedColumnLayer TypeError

### ISSUE THREAD

**Issue by miccferr** [keplergl/kepler.gl#1644]
**Describe the bug**
Kepler fails to load an hexbin map widget with error:

 `deck: error during initialisation of EnhancedColumnLayer({id: 'lifibkk-hexagon-cell'}) TypeError: Class constructor o cannot be invoked without 'new'`

**To Reproduce**
1. load kepler
2. Add data: `GeoDataFrame` with `hex_id` column containing H3 hashes at resolution 5,6
3. Try to load map

**Expected behavior**

An interactive map widget should be displayed. It works at resolution 5, but fails for hashes at resolution 6.

**Screenshots**

**Environment (please complete the following information):**
 - Python version: 3.7.7
 - keplergl Widget version: 0.3.2

**Additional context**
I tried loading the same data on the [demo website](https://kepler.gl/demo) and it works. 

Within Jupyter, I also converted the hashes to lat/lon points and they are correctly loaded in the expected location

**Comment by shern2** [comment]
Thank you for this useful tool. 
I have a similar error.
Have tested all the notebooks in https://github.com/keplergl/kepler.gl/tree/master/bindings/kepler.gl-jupyter/notebooks
**All worked, except for the hexbin one.**

**Environment**
python: 3.9.7
Keplergl: 0.3.2
node: v14.13.0

**Error**
Chrome console shows errors like:
> deck: error during initialization of EnhancedColumnLayer({id: 'jdys7lp-hexagon-cell'}) TypeError: Class constructor o cannot be invoked without 'new'
> TypeError: Class constructor o cannot be invoked without 'new'

The error suggests it's related to the `deck.gl` version issues 
e.g.
https://github.com/keplergl/kepler.gl/issues/1601
But am confused why it's still occurring, since the fix seems to be fixing the version to deck.gl 8.2.0 as per https://github.com/keplergl/kepler.gl/pull/1602

Hope you can assist.
Thanks!

**Comment by shern2** [comment]
Update: Surprisingly In Google Colab, hexbin shows up correctly..:

![Screenshot from 2021-12-28 16-28-31](https://user-images.githubusercontent.com/18212522/147545825-8e0181af-3a06-468b-b7dd-fefd6cea117d.png)

**Colab env**
python: 3.7.12
Keplergl: 0.3.2
node: v14.16.0

**Comment by Sandbird** [comment]
When you save_to_html though, the example works. I think this is the reason why it's not working.
The call to mapbox.com is not getting an answer back...probably because there is nothing running at http://localhost:8888.
While the html way..sends a Null as origin.


![Screenshot_1](https://user-images.githubusercontent.com/204934/156560705-b6eacf63-40e4-47f9-8a59-441ecdc6592f.png)
![Screenshot_2](https://user-images.githubusercontent.com/204934/156560710-a8166877-b028-4798-968b-2ddea24bd310.png)

**Comment by JanKleine** [comment]
Hi,

Is there an update on this? I have the same issue, both in jupyter in the browser and in VSCode.

To give a minimal example, I was able to reproduce the issue with the example from the [kepler-jupyter README](https://github.com/keplergl/kepler.gl/blob/master/docs/keplergl-jupyter/README.md#3-data-format):

```python
csvData = "hex_id,value\n89283082c2fffff,64\n8928308288fffff,73\n89283082c07ffff,65\n89283082817ffff,74\n89283082c3bffff,66"

map = KeplerGl()
map.add_data(data=csvData, name="data")
map
```

I was also able to get these error messages by loading the data but not displaying them to a layer yet (by overwriting the config) and then manually creating a layer with the data :
<img width="362" alt="image" src="https://user-images.githubusercontent.com/16564943/168121531-f922c261-f7eb-4316-85b0-48bbf7de2252.png">

**Comment by inmesan** [comment]
Same here. Please, any update on this? Any older version working OK?

**Comment by inmesan** [comment]
However, the Demo version at kepler.gl works fine...

**Comment by sinannoureddine** [comment]
Is there any update on this?

That's weird that's a very common issue and it hasn't been resolved yet

**Comment by Sandbird** [comment]
I dont remember how i solved this...but at some point i was looking at the accesstoken key.
I think the dark theme is under a private token. Try creating a new token at mapbox.com and use that accesstoken instead.
Maybe create the same dark theme as well.

**Comment by sinannoureddine** [comment]
Could you please elaborate more? Can you share your package.json file

**Comment by Sandbird** [comment]
Hmm looking at it now i think I solved it differently. [https://pastebin.com/raw/uZQqJJXj](https://pastebin.com/raw/uZQqJJXj). 
I uploaded my html file here. Save it somewhere and then search for: **const MAPBOX_TOKEN** and just paste there an access token from mapbox.com (it's free, just create an account there and you can get one.)
Save and load the file on a browser and you'll see on the left my dataset...using hexbins.
So basically i just uploaded my dataset and manipulated it on the map then exported it to a single file.
IF i am not mistaken this is what happens with the demo (if you see further up i posted a picture with the problem back then). The demo uses an accesstoken that doesnt work locally..that's why i did it this way.
![image](https://user-images.githubusercontent.com/204934/180660601-c2295b5c-4b2b-4d4f-af18-f501166d1cac.png)

**Comment by sinannoureddine** [comment]
> Hmm looking at it now i think I solved it differently. https://pastebin.com/raw/uZQqJJXj.
> I uploaded my html file here. Save it somewhere and then search for: **const MAPBOX_TOKEN** and just paste there an access token from mapbox.com (it's free, just create an account there and you can get one.)
> Save and load the file on a browser and you'll see on the left my dataset...using hexbins.
> So basically i just uploaded my dataset and manipulated it on the map then exported it to a single file.
> IF i am not mistaken this is what happens with the demo (if you see further up i posted a picture with the problem back then). The demo uses an accesstoken that doesnt work locally..that's why i did it this way.
> ![image](https://user-images.githubusercontent.com/204934/180660601-c2295b5c-4b2b-4d4f-af18-f501166d1cac.png)

Thanks a lot @Sandbird ! That actually helped !

**Comment by sinannoureddine** [comment]
Thanks a lot @Sandbird ! That actually helped!

**Comment by boxinqb** [comment]
Are there alternative solutions that we can solve the issue? :)

**Comment by Sandbird** [comment]
Best guess....get an accesstoken from mapbox first, use your browser console to see which accesstoken your map uses in jupiter, search for that string in the source files (keplergl), and replace it with your own. I know it's a very 'hacky' way...but IF this is the reason why it is not working inside jupiter...this should fix the problem.

**Comment by boxinqb** [comment]
Thank you for the suggestion! :) @Sandbird

**Comment by SebastianoF** [comment]
It seems to be a problem with some hexagons, and not related to the mapbox key.

Here is an example to reproduce the bug:

```
    import pandas as pd  # 1.5.1
    from keplergl import KeplerGl  # 0.3.2

    # This fake dataset does not work, and provide a white screen.
    df_fake = pd.DataFrame({"hex_id": ["8bbe08421b1afff", "8bbe08424320fff"], "num": [2,4]})  
    
    # This fake dataset works correctly (comment it out to try the faulty one):
    df_fake = pd.DataFrame({"hex_id": ["85111153fffffff", "8510a49bfffffff"], "num": [2,4]}) 

    map_1 = KeplerGl(
        data={
            "hex_id": df_fake.copy()
        }, 
    )
    map_1

```
Hope this help reproducing the problem.

**Comment by amonras** [comment]
same here, works at resolution 5, fails at resolution 6 and above

**Comment by hridesh2207** [comment]
Any update on this issue? It is really strange nobody is bothered about this issue? And, i could only find this thread for this specific issue; however, I believe it should be a very common issue.

Pls support to provide a solution for this issue. I cannot upload the data due to it's criticality , neither can i use kepler.gl website as it also requires me to upload the data

**Comment by hridesh2207** [comment]
Can anybody help here?

**Comment by sjgd** [comment]
Did anyone find a solution for this? I tried replacing the 'mapboxApiAccessToken' in the packages locally but it did not solve the issue.

**Comment by MFairley** [comment]
@lixun910 I see you recently updated the Python package for Kepler. Do you expect to fix this problem with H3s as part of the update? I saw the description of the update includes changing Kepler to v3 but the issue remains as far as I can tell.

**Comment by indrex** [comment]
Oh wow, this is July 2024 and I had the same issue in VSCode notebook.
